sdk/iot/core/tests/cmocka/az_iot_hub_client_tests.c

```diff
@@ -22,11 +36,267 @@ void test_az_iot_hub_client_get_default_options_succeed(void** state)
   assert_true(az_span_is_content_equal(options.user_agent, AZ_SPAN_NULL));
 }
 
+void test_az_iot_hub_client_init_succeed(void** state)
+{
+  (void)state;
+
+  az_iot_hub_client client;
+  az_iot_hub_client_options options = az_iot_hub_client_options_default();
+  assert_int_equal(
+      az_iot_hub_client_init(&client, test_device_hostname, test_device_id, &options), AZ_OK);
+
+  assert_string_equal(TEST_DEVICE_ID_STR, az_span_ptr(client._internal.device_id));
+  assert_string_equal(TEST_DEVICE_HOSTNAME_STR, az_span_ptr(client._internal.iot_hub_hostname));
```

Comment:
You missed length check

Reply:
same for all tests

---
In reply to: [393964290](https://github.com/Azure/azure-sdk-for-c/pull/455#discussion_r393964290) [](ancestors = 393964290)

Reply:
I had made a previous comment here
https://github.com/Azure/azure-sdk-for-c/pull/471#discussion_r393354970
detailing why I haven't. If you think it might then I can. But in this case I would be testing the implementation of `az_span` not hub client.

Reply:
resolving in another PR